Makefile

```diff
@@ -29,16 +33,15 @@ all: build
 
 build: $(OBJECTS)
 
-pre:
+$(OBJDIR):
 	@mkdir -p $(OBJDIR)
 
-# Compile each of the binaries
-$(CMD_OBJECTS): pre
-	@echo [go] bin/$@
-	@go build -o ./bin/$@ -ldflags \
+$(CMD_BINS): build_cmds
+
+build_cmds: | $(OBJDIR)
```

Comment:
`|` says "this needs to exist, but it doesn't need to be older than the target. In other words, `./bin` but exist, but we don't need to rebuild anytime `./bin`'s timestamp is newer than build_cmds. As I understand it, it's commonly used for creating requisite directories.

I believe I added the `|` early on in tweaking the Makefile because without it, builds were getting triggered all the time, because `./bin` showed up as newer.

However, in the current revision, maybe it's not necessary since `build_cmds` doesn't actually get created as a file. So in practice what happens is that the command below gets run each time, but `go install` is fast at rebuilding (so long as the ldflags didn't change).

Another version I tried was:

```
$(CMD_BINS): | $(OBJDIR)
  GOBIN=$(OBJDIR) go install ...
```
Unfortunately when there was an updated file, Make wasn't smart enough to know if could run the `go install` command just once, and it run `go install` N times, once for each member of `$(CMD_BINS)`. Ideally I'd like Make to know it doesn't have to even run `go install` if `$(CMD_BINS)` are newer than the source, but I'm not sure how (short of touching an intermediate file, which seemed like too much complication).